src/instructlab/model/list.py

```diff
@@ -0,0 +1,138 @@
+# SPDX-License-Identifier: Apache-2.0
+
+# Standard
+from pathlib import Path
+import logging
+import os
+import time
+
+# Third Party
+import click
+import pandas
+
+# First Party
+from instructlab import clickext
+from instructlab.configuration import DEFAULTS
+from instructlab.model.backends.backends import is_model_gguf, is_model_safetensors
+
+logger = logging.getLogger(__name__)
+
+
+@click.command(name="list")
+@clickext.display_params
+@click.option(
+    "--model-dirs",
+    help="Base directories where models are stored.",
+    multiple=True,
+    default=[DEFAULTS.MODELS_DIR],
+    show_default=True,
+)
+@click.option(
+    "--list-checkpoints",
+    is_flag=True,
+)
+def model_list(model_dirs: list[str], list_checkpoints: bool):
+    """lists models"""
+    data = []
+    for directory in model_dirs:
+        for entry in Path(directory).iterdir():
+            # if file, just tally the size. This must be a GGUF.
+            if entry.is_file() and is_model_gguf(entry):
+                model, age, size = _analyze_gguf(entry)
+                data.append([model, age, size])
+            elif entry.is_dir():
+                for m in _analyze_dir(entry, list_checkpoints, directory):
+                    data.append(m)
+    print_table(["Model Name", "Model Age", "Model Size"], data)
+
+
+# convert_bytes_to_proper_mag takes a dir/file size in Bytes and converts it to the proper Magnitude (MB, GB)
+def convert_bytes_to_proper_mag(f_size: float):
+    magnitudes = ["KB", "MB", "GB"]
+    magnitude = "B"
+    for mag in magnitudes:
+        if f_size >= 1024:
+            magnitude = mag
+            f_size /= 1024
+        else:
+            return f_size, magnitude
+    return f_size, magnitude
+
+
+# print_table displays the models on the system in a nicely formatted table
+def print_table(headers: list[str], data: list[list[str]]):
```

Comment:
I'm not sure how to suggest this big of a change and don't feel like making a separate PR for this, but I've fixed up how we're printing this so now it's printing in the format we originally wanted from PrettyTable:

```
+--------------------------------+---------------------------+-----------------------+
| Model Name                     | Model Age                 | Model Size            |
+--------------------------------+---------------------------+-----------------------+
| merlinite-7b-lab-Q4_K_M-2.gguf | 0 days 00:35:09.434472546 | 4.068468689918518 GB  |
| instructlab/granite-7b-lab     | 0 days 00:32:32.872044662 | 12.553834554739296 GB |
| merlinite-7b-lab-Q4_K_M.gguf   | 0 days 00:40:45.984725562 | 4.068468689918518 GB  |
+--------------------------------+---------------------------+-----------------------+
```


```suggestion
# print_table displays the models on the system in a nicely formatted table
def print_table(headers: list[str], data: list[list[str]]):
    column_widths = [
        max(len(str(row[i])) for row in data + [headers]) for i in range(len(headers))
    ]

    # Print separator line between headers and data
    horizontal_lines = ["-" * (width + 2) for width in column_widths]
    joining_line = "+" + "+".join(horizontal_lines) + "+"

    print(joining_line)

    outputs = []
    for header, width in zip(headers, column_widths, strict=False):
        outputs.append(f" {header:{width}} ")
    print("|" + "|".join(outputs) + "|")

    print(joining_line)

    for row in data:
        outputs = []
        for item, width in zip(row, column_widths, strict=False):
            outputs.append(f" {item:{width}} ")
        print("|" + "|".join(outputs) + "|")

    print(joining_line)```

Reply:
thanks, incorporated

Reply:
this is REALLY nice